XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -1083,27 +1079,64 @@ private void generateDisconnectedHostBox()
 
         }
 
+        private string GetCertificateType(certificate_type typ)
+        {
+            switch (typ)
+            {
+                case certificate_type.ca:
+                    return Messages.CERTIFICATE_TYPE_CA;
+                case certificate_type.host:
+                    return Messages.CERTIFICATE_TYPE_HOST;
+                case certificate_type.host_internal:
+                    return Messages.CERTIFICATE_TYPE_HOST_INTERNAL;
+                case certificate_type.unknown:
+                default:
+                    return Messages.UNKNOWN;
+            }
+        }
+
         private void GenerateCertificateBox()
         {
-            if (xenObject is Host host && Helpers.StockholmOrGreater(host) && host.certificates != null && host.certificates.Count > 0)
+            if (xenObject is Host host && Helpers.StockholmOrGreater(host) && host.certificates != null)
             {
-                var certificate = host.Connection.Resolve(host.certificates[0]);
-                if (certificate == null)
-                    return;
+                var certificates = host.certificates.Select(c => host.Connection.Resolve(c)).Where(c => c != null).ToList();

```

Comment:
Is there a way to sort the certificated by type, so they are always dispalyed in the same order, e.g. identity certificate first, then internal identity certificate?